vignettes/custom_epiworkflows.Rmd

```diff
@@ -0,0 +1,598 @@
+
+## Define the `frosting()`
+
+The post-processing `frosting` layers[^1] found in `four_week_ahead` look like:
+
+```{r inspect_fwa_layers, warning=FALSE}
+epipredict::extract_frosting(four_week_ahead$epi_workflow)
+```
+
+Note: since `frosting` is unique to this package, we've defined a custom function `extract_frosting()` to inspect these steps.
+
+Using the detailed information in the output above,
+we can recreate the layers similar to how we defined the
+`recipe` `step`s[^2]:
+
+```{r make_frosting}
+four_week_layers <- frosting() |>
+  layer_predict() |>
+  layer_residual_quantiles(quantile_levels = c(0.1, 0.25, 0.5, 0.75, 0.9)) |>
+  layer_add_forecast_date() |>
+  layer_add_target_date() |>
+  layer_threshold()
+```
+
+`layer_predict()` needs to be included in every postprocessor to actually train the model.
+
+Most layers work with any engine or `step`s.
+There are a couple of layers, however, which depend on whether the engine predicts quantiles or point estimates.
+
+The layers that are only supported by point estimate engines (such as
+`linear_reg()`) are:
+
+- `layer_residual_quantiles()`: the preferred method of generating quantiles for
+  models that don't generate quantiles themselves.
+  This function uses the error residuals of the engine to calculate quantiles. 
+  This will work for most `{parsnip}` engines.
+- `layer_predictive_distn()`: alternate method of generating quantiles using
+  an approximate parametric distribution. This will work for linear regression
+  specifically.
+  
+<!-- TODO check this -->
+  
+On the other hand, the layers that are only supported by quantile estimating
+engines (such as `quantile_reg()`) are
+
+- `layer_quantile_distn()`: adds the specified quantiles.
+  If they differ from the ones actually fit, they will be interpolated and/or
+  extrapolated.
+- `layer_point_from_distn()`: this adds the median quantile as a point estimate,
+  and, if called, should be included after `layer_quantile_distn()`. 
+
+## Fitting an `epi_workflow()`
+
+Now that we have a recipe and some layers, we can assemble the workflow.
+This is as simple as passing the component preprocessor, model, and postprocessor into `epi_workflow()`.
+
+```{r workflow_building}
+four_week_workflow <- epi_workflow(
+  four_week_recipe,
+  linear_reg(),
+  four_week_layers
+)
+```
+
+After fitting it, we will have recreated `four_week_ahead$epi_workflow`.
+
+```{r workflow_fitting}
+fit_workflow <- four_week_workflow |> fit(training_data)
+```
+
+Running `fit()` calculates all preprocessor-required parameters, and trains the model on the data passed in `fit()`.
+However, it does not generate any predictions; predictions need to be created in a separate step.
+
+## Predicting
+
+To make a prediction, we need to first narrow a data set down to the relevant observations.
+This process removes observations that will not be used for training because, for example, they contain missing values or <!-- TODO other reasons?-->.
```

Comment:
```suggestion
To make a prediction, it helps to narrow the data set down to the relevant observations using `get_test_data()`. Not doing this will still fit, but it will predict on every day in the data-set, and not just on the `forecast_date`.
```

Here's a possible rephrasing